### PR TITLE
Refactor Server classes.

### PR DESCRIPTION
**Pull request by tpazderka**
- [x] Any changes relevant to users are recorded in the `CHANGELOG.md`.
- [ ] The documentation has been updated, if necessary.
---
A touch of refactoring. Pruned Server classes.

### CODE REVIEW

**Comment by tpazderka** [comment]
Part of #629

File: src/oic/oic/__init__.py [hunk 1]
```diff
@@ -1454,9 +1452,9 @@ def _parse_urlencoded(url=None, query=None):
 
         return parse_qs(query)
 
-    def parse_token_request(self, request=AccessTokenRequest,
-                            body=None):
-        return oauth2.Server.parse_token_request(self, request, body)
+    def parse_token_request(self, request=AccessTokenRequest, body=None):
+        """Overriden to use OIC Message type."""
```
Comment (schlenk) [inline src/oic/oic/__init__.py]: Overridden

File: src/oic/oic/__init__.py [hunk 2]
```diff
@@ -1570,15 +1568,16 @@ def parse_authorization_request(self, request=AuthorizationRequest,
...
-
-        return oauth2.Server.parse_jwt_request(self, request, txt, keys, verify,
-                                               sender=sender)
-
-    def parse_refresh_token_request(self,
-                                    request=RefreshAccessTokenRequest,
-                                    body=None):
-        return oauth2.Server.parse_refresh_token_request(self, request, body)
+                          keyjar=None, verify=True, sender='', **kwargs):
+        """Overriden to use OIC Message type."""
```
Comment (schlenk) [inline src/oic/oic/__init__.py]: Overridden

File: src/oic/oic/__init__.py [hunk 3]
```diff
@@ -1570,15 +1568,16 @@ def parse_authorization_request(self, request=AuthorizationRequest,
...
+        if 'keys' in kwargs:
+            keyjar = kwargs['keys']
+            warnings.warn('`keys` are renamed to `keyjar`, please update your code.', DeprecationWarning)
```
Comment (schlenk) [inline src/oic/oic/__init__.py]: `keys` was renamed ...

File: src/oic/oic/__init__.py [hunk 4]
```diff
@@ -1570,15 +1568,16 @@ def parse_authorization_request(self, request=AuthorizationRequest,
...
+        return super().parse_jwt_request(request=request, txt=txt, keyjar=keyjar, verify=verify, sender=sender)
+
+    def parse_refresh_token_request(self, request=RefreshAccessTokenRequest, body=None):
+        """Overriden to use OIC Message type."""
```
Comment (schlenk) [inline src/oic/oic/__init__.py]: Overridden

**Comment by tpazderka** [comment]
The typos were fixed.

**Pull request merged**